app/beacon.py

```diff
@@ -41,23 +50,40 @@ def get_actual_slot(self):
         actual_slots['finalized_slot'] = int(response['data']['header']['message']['slot'])
         return actual_slots
 
-    def get_balances(self, epoch, key_list):
-        payload = {}
+    def _convert_key_list_to_str_arr(self, key_list):
         pubkeys = []
-        slot = epoch * self.slots_per_epoch
         for key in key_list:
             pubkeys.append('0x' + binascii.hexlify(key).decode())
 
-        payload['id'] = pubkeys
-        response = requests.get(urljoin(self.url, self.api_get_balances.format(slot)))
+        return pubkeys
+
+    def get_balances(self, slot, key_list):
+        pubkeys = self._convert_key_list_to_str_arr(key_list)
+
+        logging.info(f'Fetching validators from Beacon node...')
         balance_list = []
-        for validator in response.json()['data']:
-            if validator['validator']['pubkey'] in pubkeys:
-                balance_list.append(int(validator['balance']))
+        found_on_beacon_pubkeys = []
+        for pubkey in pubkeys:
+            json = requests.get(
+                urljoin(self.url, self.api_get_balance.format(slot, pubkey))
+            ).json()
+
+            if 'data' in json:
+                balance_list.append(int(json['data']['balance']))
+                found_on_beacon_pubkeys.append(json['data']['validator']['pubkey'])
+
+        # Log all validators along with balance
+        logging.info(f'Validator balances on beacon for slot: {slot}')
+        for pubkey, balance in zip(found_on_beacon_pubkeys, balance_list):
+            logging.info(f'Pubkey: {pubkey[:12]} Balance: {balance} Gwei')
         balances = sum(balance_list)
+        logging.info(f'Validator balances sum: {balances} Gwei')
+
         # Convert Gwei to wei
         balances *= 10 ** 9
-        return balances
+        total_validators_on_beacon = len(found_on_beacon_pubkeys)
+
+        return balances, total_validators_on_beacon
```

Comment:
Prysm is still broken right now since `Prysm.get_balances` should return two values: `balances` and `total_validators_on_beacon`, like `Lighthouse.get_balances` does. But `Prysm.get_balances` only returns `balances` in this branch. Have you tried running it with Prysm before committing https://github.com/lidofinance/lido-oracle/pull/38/commits/9e703b97e58c4efe129ed9e383224b47a3ca5cfa#diff-c54a407f57e30d59dbaffdba7dc0bb249b021ff7de1073d7d94ab525b022ff5fL20?